### PR TITLE
monitoring-plugins: 2.3.5 -> 2.4.0

### PR DESCRIPTION
**Pull request by oxzi**
## Description of changes

> https://github.com/monitoring-plugins/monitoring-plugins/releases/tag/v2.4.0

All patches are now included in this release and could been removed.

As upstream removed the translations, the Makefile does not need to be patched anymore[^0].

The license was fixed, as it is GPL-3.0-or-later, take a look at the "or (at your option) any later version" part in each license header[^1].

[^0]: https://github.com/monitoring-plugins/monitoring-plugins/commit/40370b8ff0332692c1d8d47c24ebda02b7a18823
[^1]: https://github.com/monitoring-plugins/monitoring-plugins/commit/fa4efcdf6df1096ff1f6740c7f44b213b47dc9d3

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by oxzi** [comment]
@anthonyroussel: As you just merged #298757, addressing similar topics, would you please take a look? Thanks :)

File: pkgs/servers/monitoring/plugins/default.nix [hunk 1]
```diff
@@ -76,9 +60,6 @@ stdenv.mkDerivation rec {
   # --with-ping-command needs to be done here instead of in
   # configureFlags due to the spaces in the argument
   postPatch = ''
-    substituteInPlace po/Makefile.in.in \
-      --replace /bin/sh ${runtimeShell}
-
     sed -i configure.ac \
       -e 's|^DEFAULT_PATH=.*|DEFAULT_PATH=\"${binPath}\"|'
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/plugins/default.nix:65]: the configureFlagsArray can be merged into configureFlags

Comment (oxzi) [reply pkgs/servers/monitoring/plugins/default.nix:65]: As the comment says, this does not work due to the spaces.

https://github.com/NixOS/nixpkgs/blob/d17488699f04a0a04ee4402ff22275c384508a9e/pkgs/servers/monitoring/plugins/default.nix#L60-L61

I have just verified this. The comment is still correct.

**Review by anthonyroussel — APPROVED** [review]
Looks good to me :heavy_check_mark: 
(Sorry for the late review, I hadn't seen the PR in the notification pages)

* Diff: https://github.com/monitoring-plugins/monitoring-plugins/compare/v2.3.5...v2.4.0
* Build OK on x86_64-linux :heavy_check_mark: 
* Tested some `check_` binaries :heavy_check_mark:

**Comment by oxzi** [comment]
Thanks for taking the time and merging it, @anthonyroussel.

**Pull request merged**